lib/logstash/filters/grok.rb

```diff
@@ -288,21 +288,26 @@ def filter(event)
         end
         #break if done
       end # @patterns.each
-  
+
       if matched
         filter_matched(event)
       else
         # Tag this event if we can't parse it. We can use this later to
         # reparse+reindex logs if we improve the patterns given.
         @tag_on_failure.each do |tag|
-          event["tags"] ||= []
-          event["tags"] << tag unless event["tags"].include?(tag)
+          # do not replace the code below with:
```

Comment:
I think it's safe to remove this warning, it makes sense in the example below where a developer will instinctively try to optimize/dry/refactor code, here I doubt someone would replace `event.tag(tag)` with 2 lines of more complex code